src/compat/posix/fs/readv.c

```diff
@@ -0,0 +1,49 @@
+#include <unistd.h>
+#include <errno.h>
+
+#include <fs/index_descriptor.h>
+#include <fs/idesc.h>
+#include <kernel/task/resource/idesc_table.h>
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+#include <util/macro.h>
+#include <kernel/task/resource/idesc_table.h>
+#include <net/sock.h>
+#include <net/socket/socket_desc.h>
+#include <net/socket/ksocket.h>
+
+#define  socket_idesc_check(sockfd, sk) \
```

Comment:
How about to extract this repetative macro definition to header file?
I'd suppose to place it near these .c files and include like  `#include "vec_common.h"` or something like this